Inc/HALAL/Models/Packets/SPIPacket.hpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * SpiPacket.hpp
+ *
+ *  Created on: 15 nov. 2023
+ *      Author: ricardo
+ */
+#pragma once
+
+#include "ErrorHandler/ErrorHandler.hpp"
+
+
+class SPIPacket{
+public:
+	static map<uint16_t, SPIPacket*> SPIPacketsByID;
+	uint16_t id;
+	uint8_t* master_data;
+	uint16_t master_data_size;
+	uint8_t* slave_data;
+	uint16_t slave_data_size;
+	uint16_t greater_data_size;
+
+	/**
+	 * @brief constructor of SPIPacket. The SPIPacket uses the first byte of the data arrays to coordinate the packet id, and builds the arrays with 1 more byte of space.
+	 *
+	 */
+	SPIPacket(uint16_t id, uint16_t master_data_size, uint16_t slave_data_size) : id(id), master_data_size(master_data_size), slave_data_size(slave_data_size){
+		if(id == 0){
+			ErrorHandler("Used reserved ID 0 on SPIPacket");
```

Comment:
clarify ErrorHandler message, smth like "0 is a reserved ID "

Reply:
👍